x-pack/test/spaces_api_integration/common/fixtures/es_archiver/saved_objects/spaces/data.json

```diff
@@ -120,7 +120,9 @@
       }],
       "type": "dashboard",
       "updated_at": "2017-09-21T18:49:16.270Z",
-      "namespaces": ["default"]
+      "namespaces": ["default"],
+      "coreMigrationVersion": "8.8.0",
+      "typeMigrationVersion": "8.7.0"
```

Comment:
Could you explain if specific versions pair affects the way migration is run?

Reply:
They do, indeed.

For this particular case I took the latest versions available for 8.8.0. As you can see they depend on the type, e.g.:
* for dashboards the latest `typeMigrationVersion` is `8.7.0`
* for visualizations, it is `8.5.0`

These properties dictate the way the migration logic behaves when upgrading.
In particular, there's an "_outdated documents search + transform_" loop, that takes any documents that are not on the latest version for either of these 2 properties, and transforms them.

We have some transformations that are defined by core, and others that are defined by plugin owners.

E.g. if the owners of `'visualization'` SO type decide that they have to transform their stored SOs in 8.10.0, they will define a `'8.10.0': () => {} // transform` function.
The migration logic will run these `transform` functions at startup, and update the property accordingly.

Thus, these properties should match the versions that were being used at the time when the `data.json` was created. This way, if there are transforms defined after that point in time, the documents will be updated accordingly.

That being said, I took the latest, and the `data.json` at stakes were probably aligned with older versions.
They seem to work, but I wasn't really accurate here.

I believe if the fields are not defined, the documents will then go through all of the defined transforms. TBC

Reply:
**UPDATE:** if the fields are not defined, the documents won't go through any of the defined transforms.